extension/src/shared/view-layers/rootLayer.ts

```diff
@@ -120,7 +120,11 @@ const findSetting = (settings: Setting[], currentDataId: string | undefined) =>
   const hasGroups = fieldComponents?.groups?.some(g => !!g.components.length);
   const hasTemplate = fieldComponents?.useTemplate && !!fieldComponents.templateId;
 
-  return hasGroups || hasTemplate ? setting : defaultSetting;
+  return {
+    ...defaultSetting,
+    ...setting,
+    fieldComponents: hasGroups || hasTemplate ? fieldComponents : defaultSetting?.fieldComponents,
```

Comment:
Do we still need this condition? I thought we can do like below.
```
return {
  ...defaultSetting,
  ...setting,
};
```

Reply:
I think it still has some difference like if we have groups = [] or useTemplate = true but templateId = undefined in `fieldComponents`, it should use defaultSettings, without the conditions it will use this empty array or use template with no id.